File: CSE241 - Object Oriented Programming/studies/cpp/ch14-Inheritance/employee/salariedEmployee.cpp
#include <iostream>
#include "employee.h"
#include "salariedEmployee.h"

using std::cerr;
using std::cout;
using std::endl;
using std::string;

namespace EmployeeGTU {
    SalariedEmployee::SalariedEmployee () 
    : Employee(), _salary(0) {}

    SalariedEmployee::SalariedEmployee (const string & name, const string & ssn, double weeklySalary)
    : Employee(name, ssn), _salary(weeklySalary) {} 

    double SalariedEmployee::getSalary () const {return _salary;}
    void SalariedEmployee::setSalary (double salary) {_salary = salary;}

    // printCheck is not const because of setNetPay   
    void SalariedEmployee::printCheck() {
        setNetPay(getSalary());
        cout << "\n________________________________________________\n"
             << "Pay to the order of " << getName() << endl
             << "The sum of " << getNetPay() << " Dollars\n"
             << "_________________________________________________\n"
             << "Check Stub NOT NEGOTIABLE \n"
             << "Employee Number: " << getSsn() << endl
             << "Salaried Employee. Regular Pay: " << getSalary() << endl
             << "_________________________________________________\n";
    }
}

File: cse241-object-oriented-programming/studies/java/interfaces/payable/SalariedEmployee.java
package payable;

public class SalariedEmployee extends Employee {
    private double _weeklySalary;

    public SalariedEmployee (String fname, String lname, String ssn, double weeklySalary) {
        super(fname, lname, ssn);
        setWeeklySalary(weeklySalary);
    }

    public void setWeeklySalary (double salary) {_weeklySalary = salary < 0.0 ? 0.0 : salary;}
    
    public double weeklySalary () {return _weeklySalary;}

    public String toString () {
        return String.format("salaried employee: %s \nweekly salary: $%,.2f", 
            super.toString(), weeklySalary());
    }

    public double paymentAmount () {
        return weeklySalary();
    }
}